### ISSUE TITLE
Problem running Odalic - Null pointer exception when the task is about to start

### ISSUE THREAD

**Issue by tomas-knap** [odalic/sti#151]
I use latest develop branch (From 6.9., 14:00). 
I set up the configs as in config folder, using 3 KBs (DBpedia, DBpediaClone and DeDBpedia). Also I initiated cache folder with empty cache (from repo). 

When I define a task (using the origin Book.txt - [Books.csv.zip](https://github.com/odalic/sti/files/456815/Books.csv.zip)), run it, and there is null pointer exception - see below. Odalic Core is not executed.

@brodecva  Can you reproduce? If not, can you please send your war file?

**Comment by tomas-knap** [comment]
It seems that the root cause ( a bit hidden) is: 
```
Caused by: java.lang.NullPointerException\n\tat cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.setColumnIgnoresForInterpreter(TableMinerPlusFactory.java:130)
```

**Comment by tomas-knap** [comment]
Hm, koho zastrelim? ;-)

[https://github.com/odalic/sti/blob/develop/odalic/src/main/java/cz/cuni/mff/xrg/odalic/tasks/executions/TableMinerPlusFactory.java#L259-L261](https://github.com/odalic/sti/blob/develop/odalic/src/main/java/cz/cuni/mff/xrg/odalic/tasks/executions/TableMinerPlusFactory.java#L259-L261)

**Comment by tomas-knap** [comment]
When the proper exception handling is added:
```
.17:00:25.206 [pool-1-thread-1] INFO  c.c.m.x.o.t.e.TableMinerPlusFactory - Initializing KB cache ... 
uk.ac.shef.dcs.kbsearch.KBSearchException: Cannot proceed: the cache dir is not set or does not exist: /Users/tomasknap/Documents/PROJECTS/Odalic/sti/cache/empty/
	at uk.ac.shef.dcs.kbsearch.KBSearch.initializeSolrServer(KBSearch.java:249)
	at uk.ac.shef.dcs.kbsearch.KBSearch.getSolrServer(KBSearch.java:96)
	at uk.ac.shef.dcs.kbsearch.KBSearch.initializeCaches(KBSearch.java:79)
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.initComponents(TableMinerPlusFactory.java:160)
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.getInterpreters(TableMinerPlusFactory.java:107)
	at cz.cuni.mff.xrg.odalic.tasks.executions.FutureBasedExecutionService.lambda$submitForTaskId$0(FutureBasedExecutionService.java:105)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
.17:00:25.225 [pool-1-thread-1] ERROR c.c.m.x.o.t.e.TableMinerPlusFactory - uk.ac.shef.dcs.kbsearch.KBSearchException: Cannot proceed: the cache dir is not set or does not exist: /Users/tomasknap/Documents/PROJECTS/Odalic/sti/cache/empty/
	at uk.ac.shef.dcs.kbsearch.KBSearch.initializeSolrServer(KBSearch.java:249)
	at uk.ac.shef.dcs.kbsearch.KBSearch.getSolrServer(KBSearch.java:96)
	at uk.ac.shef.dcs.kbsearch.KBSearch.initializeCaches(KBSearch.java:79)
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.initComponents(TableMinerPlusFactory.java:160)
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.getInterpreters(TableMinerPlusFactory.java:107)
	at cz.cuni.mff.xrg.odalic.tasks.executions.FutureBasedExecutionService.lambda$submitForTaskId$0(FutureBasedExecutionService.java:105)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
 
uk.ac.shef.dcs.sti.STIException: Failed initialising KBSearch cache.
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.initComponents(TableMinerPlusFactory.java:165)
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.getInterpreters(TableMinerPlusFactory.java:107)
	at cz.cuni.mff.xrg.odalic.tasks.executions.FutureBasedExecutionService.lambda$submitForTaskId$0(FutureBasedExecutionService.java:105)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: uk.ac.shef.dcs.kbsearch.KBSearchException: Cannot proceed: the cache dir is not set or does not exist: /Users/tomasknap/Documents/PROJECTS/Odalic/sti/cache/empty/
	at uk.ac.shef.dcs.kbsearch.KBSearch.initializeSolrServer(KBSearch.java:249)
	at uk.ac.shef.dcs.kbsearch.KBSearch.getSolrServer(KBSearch.java:96)
	at uk.ac.shef.dcs.kbsearch.KBSearch.initializeCaches(KBSearch.java:79)
	at cz.cuni.mff.xrg.odalic.tasks.executions.TableMinerPlusFactory.initComponents(TableMinerPlusFactory.java:160)
	... 6 more

```

**Comment by brodecva** [comment]
> When the proper exception handling is added:

@tomas-knap Somehow I get a feeling that we have already encountered this problem when finishing the 1st iteration and the fix was somehow forgotten.

Concerning the obscured exception, the fault is all mine. I should have checked the implementation more rigorously later. I have accepted @josef-janousek solution, which was heavily based on the existing batch construction done by Ziqi Zhang. It was perfectly reasonable at the time as we had little understanding (and in many cases still have) of the TableMiner+ interpreter order of construction and its dependencies and it worked well for the prototype. Hopefully we will be able to clean up the code in spare time while doing the planned re-factoring of the core. Unfortunately this 'sin' got overlooked. Sorry about the wasted time.
